test_chess.py

```diff
@@ -0,0 +1,105 @@
+import pytest
+from unittest.mock import Mock, patch
+from chess_engine import game_state
+from Piece import Knight, Bishop
+from enums import Player
+import ai_engine
+
+
+class TestChess:
+    @pytest.fixture
+    def game(self):
+        return game_state()
+
+    @pytest.fixture
+    def empty_the_board(self):
+        game = game_state()
+        empty_row = [Player.EMPTY] * 8
+        game.board = [empty_row[:] for _ in range(8)]
+        return game
+
+    # knight in the center
+    def test1_get_valid_peaceful_moves(self, game):
+        knight = Knight('n', 3, 4, Player.PLAYER_1)
+        with patch.object(game, 'get_piece', return_value=Player.EMPTY):
+            moves = knight.get_valid_peaceful_moves(game)
+            correct_moves = [(1, 3), (2, 2), (4, 2), (5, 3), (5, 5), (4, 6), (2, 6), (1, 5)]
+            assert set(moves) == set(correct_moves)
+
+    # knight in the corner
+    def test2_knight_get_valid_peaceful_moves(self, game):
+        knight = Knight('n', 0, 0, Player.PLAYER_1)
+
+        def mock_get_piece(r, c):
+            if (r, c) != (0, 0) and (0 <= r < 8) and (0 <= c < 8):
+                return Player.EMPTY
+            else:
+                return None
+
+        game.get_piece = Mock(side_effect=mock_get_piece)
+        returned_value = knight.get_valid_peaceful_moves(game)
+        correct_value = [(2, 1), (1, 2)]
+        assert set(returned_value) == set(correct_value)

```

Comment:
the terms can be a bit clearer -
returned_value = actual_values
correct_value = expected_values.
make your variables the most indicative you can.